extensions/ql-vscode/src/compare-performance/compare-performance-view.ts

```diff
@@ -94,4 +118,287 @@ export class ComparePerformanceView extends AbstractWebview<
         break;
     }
   }
+
+  async downloadExternalLogs(): Promise<void> {
+    const client = await this.app.credentials.getOctokit();
+    async function getArtifactDownloadUrl(
+      url: string,
+    ): Promise<{ url: string; bytes: number; id: string }> {
+      const pattern =
+        /https:\/\/github.com\/([^/]+)\/([^/]+)\/actions\/runs\/([^/]+)\/artifacts\/([^/]+)/;
+      const match = url.match(pattern);
+      if (!match) {
+        throw new Error(`Invalid artifact URL: ${url}`);
+      }
+      const [, owner, repo, , artifact_id] = match;
+      const response = await client.request(
+        "HEAD /repos/{owner}/{repo}/actions/artifacts/{artifact_id}/{archive_format}",
+        {
+          owner,
+          repo,
+          artifact_id,
+          archive_format: "zip",
+        },
+      );
+      if (!response.headers["content-length"]) {
+        throw new Error(
+          `No content-length header found for artifact URL: ${url}`,
+        );
+      }
+      return {
+        url: response.url,
+        bytes: response.headers["content-length"],
+        id: `artifacts/${owner}/${repo}/${artifact_id}`,
+      };
+    }
+
+    const downloadLog = async (originalUrl: string) => {
+      const {
+        url,
+        bytes,
+        id: artifactDiskId,
+      } = await getArtifactDownloadUrl(originalUrl);
+      const logPath = path.join(
+        this.workingDirectory,
+        `logs-of/${artifactDiskId}`,
+      );
+      if (existsSync(logPath) && readdirSync(logPath).length > 0) {
+        void extLogger.log(
+          `Skipping log download and extraction to existing '${logPath}'...`,
+        );
+      }
+      await withProgress(
+        async (progress) => {
+          const downloadPath = path.join(this.workingDirectory, artifactDiskId);
+          if (
+            existsSync(downloadPath) &&
+            readdirSync(downloadPath).length > 0
+          ) {
+            void extLogger.log(
+              `Skipping download to existing '${artifactDiskId}'...`,
+            );
+          } else {
+            await ensureDir(downloadPath);
+            void extLogger.log(
+              `Downloading from ${artifactDiskId} (bytes: ${bytes}) ${downloadPath}...`,
+            );
+            await this.fetchAndUnzip(url, downloadPath, progress);
+          }
+          if (existsSync(logPath) && readdirSync(logPath).length >= 0) {
```

Comment:
```suggestion
          if (existsSync(logPath) && readdirSync(logPath).length > 0) {
```
`.length >= 0` is always true.